core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -148,7 +148,10 @@ public void trackUpdatedKey(int segment, Object key, Object lockOwner) {
       if (log.isTraceEnabled()) {
          log.tracef("Tracking key for %s: %s", lockOwner, key);
       }
-      updatedKeys.put(key, new State(segment, key, lockOwner));
+      State old = updatedKeys.put(key, new State(segment, key, lockOwner));
+      if (old != null) {
+         old.discard(); //avoid sending the cleanup command to the cluster members
```

Comment:
Nitpicking, but IMO the comment is too long, it should be on its own line (above)